storage.go

```diff
@@ -218,14 +218,17 @@ func (storage *MongoStorage) UpdateJob(job *SubmittedJob) error {
 // GetAccount loads an account by its unique account name, creating it if it doesn't already exist.
 func (storage *MongoStorage) GetAccount(name string) (*Account, error) {
 	out := Account{Name: name}
-	_, err := storage.accounts().FindId(name).Apply(mgo.Change{
-		Update:    bson.M{"$setOnInsert": out},
-		Upsert:    true,
-		ReturnNew: true,
-	}, &out)
-	if err != nil {
+	var acct Account
+	err := storage.accounts().FindId(name).One(&acct)
+	if err != nil && err != mgo.ErrNotFound {
 		return nil, err
 	}
+	if err == mgo.ErrNotFound {
+		err = storage.accounts().Insert(out)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Dumb question - doesn't this introduce a race condition? It seems to me that it's possible another service could create an account with the same name in between the `FindId` call and the `Insert` call.

If the upsert isn't an option with Mongo 2.6, maybe we could check the `err` returned by `Insert` and re-query if it's indicating a duplicate `_id`?